crates/cli-support/src/js/closures.rs

```diff
@@ -226,26 +226,32 @@ impl ClosureDescriptors {
 
             let (js, _ts, _js_doc) = {
                 let mut builder = Js2Rust::new("", input);
+                builder.prelude("this.cnt++;");
                 if closure.mutable {
                     builder
                         .prelude("let a = this.a;\n")
                         .prelude("this.a = 0;\n")
                         .rust_argument("a")
+                        .rust_argument("b")
                         .finally("this.a = a;\n");
                 } else {
-                    builder.rust_argument("this.a");
+                    builder.rust_argument("this.a")
+                        .rust_argument("b");
                 }
+                builder.finally("if (this.cnt-- == 1) d(this.a, b);");
                 builder
                     .process(&closure.function)?
-                    .finish("function", "this.f")
+                    .finish("function", "f")
             };
             input.expose_add_heap_object();
             input.function_table_needed = true;
             let body = format!(
-                "function(ptr, f, _ignored) {{
-                    let cb = {};
-                    cb.f = wasm.__wbg_function_table.get(f);
-                    cb.a = ptr;
+                "function(a, b, fi, di, _ignored) {{
```

Comment:
What is `b` here?

Reply:
The a/b pointers here are the two parts of a trait object, with fi being the wrapper that we invoke which does wasm-bindgen translation of arguments.